sdk/Sdk.Generators/ExtensionStartupRunnerGenerator.cs

```diff
@@ -0,0 +1,212 @@
+        }
+
+        /// <summary>
+        /// Check the type to see it is valid and return an error message of the first failure.
+        /// </summary>
+        private string? GetErrorIfTypeIsNotValid(ITypeSymbol typeSymbol)
+        {
+            if (typeSymbol is INamedTypeSymbol namedTypeSymbol)
+            {
+                // Check public parameterless constructor exist for the type.
+                var constructorExist = namedTypeSymbol.InstanceConstructors
+                                                  .Any(c => c.Parameters.Length == 0 &&
+                                                            c.DeclaredAccessibility == Accessibility.Public);
+                if (!constructorExist)
+                {
+                    return $"{typeSymbol.ToDisplayString()} class must have a public parameterless constructor.";
+                }
+
+                // Check the extension startup class implements WorkerExtensionStartup abstract class.
+                if (!(namedTypeSymbol.BaseType!.GetFullName().Equals(startupBaseClassName)))
+                {
+                    return $"{typeSymbol.ToDisplayString()} must be a type which implements {startupBaseClassName}.";
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Writes an assembly attribute with type information about our auto generated WorkerExtensionStartupCodeExecutor class.
+        /// </summary>
+        private static void WriteAssemblyAttribute(IndentedTextWriter textWriter)
+        {
+            textWriter.WriteLine(
+                "[assembly: WorkerExtensionStartupCodeExecutorInfo(typeof(Microsoft.Azure.Functions.Worker.WorkerExtensionStartupCodeExecutor))]");
+        }
+
+        /// <summary>
+        /// Writes a class with code which calls the Configure method on each implementation of participating extensions.
+        /// We also have it implement the same "IWorkerExtensionStartup" interface which extension authors implement.
+        /// </summary>
+        private static void WriteStartupCodeExecutorClass(IndentedTextWriter textWriter, IEnumerable<(string TypeName, string? Error)> types)
+        {
+            textWriter.WriteLine("internal class WorkerExtensionStartupCodeExecutor : WorkerExtensionStartup");
+            textWriter.WriteLine("{");
+            textWriter.Indent++;
+            textWriter.WriteLine("public override void Configure(IFunctionsWorkerApplicationBuilder applicationBuilder)");
+            textWriter.WriteLine("{");
+            textWriter.Indent++;
+
+            foreach (var type in types)
+            {
+                textWriter.WriteLine("try");
+                textWriter.WriteLine("{");
+                textWriter.Indent++;
+                if (type.Error != null)
+                {
+                    textWriter.WriteLine($"throw new InvalidOperationException(\"{type.Error}\");");
```

Comment:
This will cause a runtime exception, which to an end user who may not be familiar with the type, could be confusing.

Should this become a build issue with information about the extension/dependency that introduced the bad startup?

Reply:
True. A build time diagnostic error makes sense. In the new iteration, I made changes to report a diagnostic error when startup implementation is invalid (missing parameter less constructor, incorrect base class).